src/server/req_quejob.c

```diff
@@ -924,8 +922,7 @@ req_quejob(struct batch_request *preq)
 		 * into an advance reservation queue, the reservation's ID
 		 * gets attached later in the code
 		 */
-		job_attr_def[(int)JOB_ATR_reserve_ID].at_decode(get_jattr(pj, JOB_ATR_reserve_ID),
-			NULL, NULL, NULL);
+		set_attr_generic(get_jattr(pj, JOB_ATR_reserve_ID), &job_attr_def[(int)JOB_ATR_reserve_ID], NULL, NULL, INTERNAL);
```

Comment:
can you use set_jattr_generic here instead?

Reply:
Well no. set_jattr_generic() will not go further since the value is NULL.